src/lib.rs

```diff
@@ -1546,3 +1469,30 @@ fn bindgen_test_layout_jpeg_memory_mgr() {
         concat!("Offset of field: ", stringify!(jpeg_memory_mgr), "::", stringify!(max_alloc_chunk))
     );
 }
+
+extern "C" {
+    #[cfg(test)]
+    fn jsimd_can_rgb_ycc() -> c_int;
+    #[cfg(test)]
+    #[allow(dead_code)]
+    fn jsimd_can_fdct_ifast() -> c_int;
+    #[cfg(test)]
+    #[allow(dead_code)]
+    fn jsimd_fdct_ifast(block: *mut DCTELEM);
+    pub fn jpeg_float_add_quant_table(
+        cinfo: &mut jpeg_compress_struct,
+        which_tbl: c_int,
+        basic_table: *const c_uint,
+        scale_factor: f32,
+        force_baseline: boolean,
+    );
+
+    /// Precalculate JPEG dimensions for current compression parameters
+    #[cfg(feature = "jpeg70_abi")]
+    pub fn jpeg_calc_jpeg_dimensions(cinfo: &mut jpeg_compress_struct);
+
+    // FIXME: wrong type (should be decompress struct, not compress),
+    // second argument is also incorrect.
+    pub fn jpeg_set_idct_method_selector(cinfo: &mut jpeg_compress_struct, param: *const c_void);
+
```

Comment:
These don't look that important, and they could be made to work, but I skipped them because I couldn't find them in headers, validation failed in multiple ways, etc. 

The generated C++ wrappers translate the Rust types to C, and then call the C intrinsics with them, so depending on the warning level set in the `cc::Builder`, stuff like mismatching `const`, etc. will trigger an error =/